### PR TITLE
Vkotha master

### PR DESCRIPTION
**Pull request by vivekkothakota**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by vivekkothakota** [comment]
Ping....
Are you waiting on me? is there something blocked from my side?

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (annatisch) [inline specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: This operation should be called `Projects_List` and the operation that lists according to resource group should be called something like `Projects_ListByResourceGroup` - would this change be possible or do you already have published SDKs that will break?
Otherwise, this would be changed to `Projects_ListBySubscription`.

Additionally there's a typo in the description: `thh`.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (annatisch) [inline specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: I don't think you can make this operation pageable - the list response has two separate list parameters.
When I try to generate a Python SDK it fails in attempting to make this operation pageable.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (annatisch) [inline specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Given the description of this operation - I would probably change the operation ID to something like "Projects_DeleteBySubscription" or "Projects_DeleteAll".
But you better understand the chosen naming convention, so up to you :)

**Review by annatisch — CHANGES_REQUESTED** [review]

**Comment by annatisch** [comment]
Hi @vivekkothakota - sorry about the late review - I requested just a couple of minor changes and we're good to go :)

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Done.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Done.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Is there a way, we can suppress this warning for now? We will make the response pageable in the next API version change.

Comment (annatisch) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Hi @vivekkothakota - this is currently an error, so it will need to be modified. When you say it will be supported in a subsequent API version, do you mean that the response will have a `nextLink`? Or do you mean that the model will be restructured to have a single iterable array?

The main reason it is failing to generate is because you need to specify the name of the array you wish to be iterated in the paged response. The response model currently defines two different arrays.
Some languages do not support multiple arrays in the response - so whichever array you select to be the iterating array, the other array will be lost during deserliazation. If you still want to do this, you need to define `itemName` as described the [x-ms-pageable docs](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable).

Alternatively, simply remove x-ms-ageable for now, and both arrays in the response will be deserilized as a normal model.

Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: thank you. remove x-ms-pageable for now..

**Review by annatisch — APPROVED** [review]

**Comment by annatisch** [comment]
@vivekkothakota - This PR is missing ARM sign-off. Since many days have passed, please let me know if you prefer to: 
1) follow up with ravbhatnagar and wait, or 
2) I can merge the PR without ARM’s team sign-off. Please note that even after merge, you should follow up with ARM team regarding ARM issues and fixes may impact the specs and/or generated SDKs.

ping @ravbhatnagar 

Thanks!

**Comment by vivekkothakota** [comment]
Thank you @annatisch. I did send an email to @ravbhatnagar, lets wait for a day for his response before merging.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Removed.

File: specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json
Comment (vivekkothakota) [reply specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json]: Done.

**Comment by annatisch** [comment]
Thanks @ravbhatnagar - looks like we're good to go? :)

**Comment by ravbhatnagar** [comment]
Looks good!

**Pull request merged**
